src/autoprops.rs

```diff
@@ -0,0 +1,261 @@
+                    if !matches!(**ty, syn::Type::Reference(_)) =>
+                {
+                    Some(quote! { let #pat = ::std::clone::Clone::clone(#pat); })
+                }
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let output = &sig.output;
+        let impl_generics_with_defaults = &sig.generics;
+
+        quote! {
+            #(#attrs)*
+            #vis fn #fn_name #impl_generics_with_defaults (#inputs) #output #where_clause {
+                #(#clones)*
+                #block
+            }
+        }
+    }
+
+    fn print_properties_struct(&self) -> proc_macro2::TokenStream {
+        let properties_name = &self.properties_name;
+        let syn::ItemFn { vis, sig, .. } = &self.item_fn;
+
+        if !self.needs_a_properties_struct() {
+            return quote! {};
+        }
+
+        let (impl_generics, type_generics, where_clause) = sig.generics.split_for_impl();
+        let fields = sig
+            .inputs
+            .iter()
+            .filter_map(|arg| match arg {
+                syn::FnArg::Typed(syn::PatType { attrs, pat, ty, .. }) => match ty.as_ref() {
+                    syn::Type::Reference(syn::TypeReference { elem, .. }) => {
+                        Some(quote! { #(#attrs)* #vis #pat: #elem, })
+                    }
+                    _ => Some(quote! { #(#attrs)* #vis #pat: #ty, }),
+                },
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let type_params = sig
+            .generics
+            .type_params()
+            .map(|param| &param.ident)
+            .collect::<Vec<_>>();
+        let phantom = (!type_params.is_empty()).then(|| {
+            quote! {
+                #[prop_or_default]
+                #vis phantom: ::std::marker::PhantomData <( #(#type_params),* )>,
+            }
+        });
+        let fields_eq = sig
+            .inputs
+            .iter()
+            .filter_map(|arg| match arg {
+                syn::FnArg::Typed(syn::PatType { pat, ty, .. }) => Some(quote_spanned! {
+                    ty.span() => self.#pat == rhs.#pat &&
+                }),
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let impl_generics_with_defaults = &sig.generics;
+
+        quote! {
+            #[derive(::yew::Properties)]
+            #vis struct #properties_name #impl_generics_with_defaults #where_clause {
+                #(#fields)*
+                #phantom
+            }
+
+            impl #impl_generics ::std::cmp::PartialEq for #properties_name #type_generics
+            #where_clause {
+                fn eq(&self, rhs: &Self) -> ::std::primitive::bool {
+                    #(#fields_eq)* true
+                }
+            }
+        }
+    }
+
+    fn needs_a_properties_struct(&self) -> bool {
+        let syn::ItemFn { sig, .. } = &self.item_fn;
+        !sig.inputs.is_empty() || sig.generics.type_params().next().is_some()
```

Comment:
```rust
fn KnownValueWrapper<T>() -> Html
where 
    T: Component,
{
    let props = props!(<T as Component>::Properties, {
        some_known_value: todo!(),
       default_config: todo!(),
      // ...
    })
    html! { <T ..props /> }
}
```
This code is valid but would be unnecessarily end up creating a properties struct

Reply:
Ah I thought it was necessary for some reason. Ok I simply removed the condition and the tests passed ff74446ed11f21efde3387e7a1d3d07eda26a33f